src/workerd/api/actor-state.h

```diff
@@ -351,11 +351,35 @@ class DurableObjectState: public jsg::Object {
   jsg::Promise<jsg::Value> blockConcurrencyWhile(jsg::Lock& js,
       jsg::Function<jsg::Promise<jsg::Value>()> callback);
 
+  void acceptWebSocket(jsg::Ref<WebSocket> ws, jsg::Optional<kj::Array<kj::String>> tags);
+  // Adds a WebSocket to the set attached to this object.
+  // `ws.accept()` must NOT have been called separately.
+  // Once called, any incoming messages will be delivered
+  // by calling the Durable Object's webSocketMessage()
+  // handler, and webSocketClose() will be invoked upon
+  // disconnect.
+  //
+  // After calling this, the WebSocket is accepted, so
+  // its send() and close() methods can be used to send
+  // messages, but its addEventListener() method won't
+  // ever receive any events as they'll be delivered to
+  // the DurableObject instead.
```

Comment:
Question on this: does this mean that the given `jsg::Ref<WebSocket>` that is passed into `acceptWebSocket` is no longer usable? Does it transition at all into a state that prevents it explicitly from being used or does it just become non-functional? What does it mean that the events will be delivered to the DurableObject? The way that is written it implies that the user would call `addEventListener('message', () => {})` on the DurableObect to receive the messages which isn't what I believe was meant? What happens if someone calls `close()` on the WebSocket after this? etc.

Reply:
> ... but I suppose we could also document it in the code?

Yeah, I think adding that additional detail to the code docs would be helpful for others coming into this later.

I do wonder about the impact of this design, however. If all I have is the `WebSocket` instance in JavaScript, and I receive it after it has been passed to `acceptWebSocket()`, how am I supposed to know that the `addEventListener()` part of the API is non-functional? For instance, suppose I have some utility from another team that adds logging to a `WebSocket` and I haven't realized that the `WebSocket` has been passed to `acceptWebSocket`? If there's no feedback when I attempt to register my event listeners then it may be rather difficult to debug.

What I'd prefer is something like...

```
const ws = // get my websocket somehow
acceptWebSocket(ws);
ws.addEventListener('message', () => {});  // throws an exception!
```

Reply:
The behavior of `addEventListener()` came up during discussion of the API, Kenton said:

> Hmm. We don't currently raise an error if you call `addEventListener()` before calling `accept()`, so e.g. you can `addEventListener()` on a WebSocket that you end up poxying through, and no events will ever be delivered. I could see a case for blocking `addEventListener()` up until `accept()` is called, but it would require a compatibility flag, and there might be some debate over whether it's allowed under the EventTarget API standard.

Might be worth bringing it up internally so we don't overcrowd the PR, I'll add clearer comments in the code though!

Reply:
I can definitely see the argument. It's one of the things I've always disliked the most about `EventTarget` (and Node.js `EventEmitter`). I certainly won't block on this point tho. We should just make sure this bit is well documented.

Reply:
Updated the comment.